### DISCUSSION TITLE
Breakpoints don't work using Platformio+VS Code "No source file named c:\...\src\main.cpp

### DISCUSSION THREAD

**Discussion by MrTucks** [earlephilhower/arduino-pico#1393] [Q&A] [answered]
Hi,
When I try to set a breakpoint in my main.cpp the following error get's shown in a popup in the bottom right corner:
No source file named c:\Users\linus\Projekte\Soundgerät\Audioversuche\TestLedBlinkt\src\main.cpp. (from break-insert "c:\\Users\\linus\\Projekte\\Soundgerät\\Audioversuche\\TestLedBlinkt\\src\\main.cpp:13")
Same error (just diffrent file of course) if I try to set breakpoints in any included libraries.

And it (obviously) doesn't stop at any breakpoints that I set there.
However, the rest of the debugging works.
And at the start of the debugging, it automatically stops at the beginning of main() at line 81 of C:\Users\linus\.platformio\packages\framework-arduinopico\cores\rp2040\main.cpp
Setting breakpoints there works.
At line 142 `setup() `of that file, I use step into it and the file setup.dbgasm is opened. I think that's the compiled version of my setup() in my main.cpp. Stepping into loop() in the same way doesn't work.

So I think it somehow doesn't know which .dbgasm's are my main.cpp.

About a month ago, it worked perfectly on my laptop. 

===Separate issue: Just writing this here because it prevents me from debugging the first issue on a second PC===
But I can't test it on my laptop again, because now uploading and debugging via picoprobe (cmsis-dap in platformio.ini) doesn't work at all on my laptop.
Error when uploading:
```
Error: unable to find a matching CMSIS-DAP device
Error: No Valid JTAG Interface Configured.
*** [upload] Error 4294967295
```
Error when trying to debug:
"Failed to launch GDB: .pioinit:13: Error in sourced command file:
Remote communication error. Target disconnected.: Sucess. (from interpreter-exec console "source .pioinit")

Many Greetings

**Comment by MrTucks** [comment]
Update on issue 2:
Windows for some/no reason switched the driver for `Picoprobe CMSIS-DAP v2(Interface 0)` from WinUSB to usbser(v1.0.0.0).
Using zadig to install the correct driver fixed this issue.
First issue is still there. It works perfectly on the laptop but not on the PC.
I will try to figure it out later.
There might be a different version of platformio, arduino-pico or some dependency, because I installed everything on the laptop about two months ago and last week on the PC.

**Comment by MrTucks** [comment, marked as answer]
I figured it out!
There was an umlaut ä somewhere in the file path of the project on the PC, but not on the laptop.
Debugging Breakpoints doesn't work if there's an Umlaut somewhere in the path!
It doesn't even build if there's an emoji or japanese character in the path.

It's kinda embarrassing, that I didn't immediately think of this. Hopefully someone else finds this post before spending hours troubleshooting. :D

**Reply by maxgerhardt** [reply]
> Debugging Breakpoints doesn't work if there's an Umlaut somewhere in the path!
>It doesn't even build if there's an emoji or japanese character in the path.

Weird, I thought PlatformIO would be free of such problems. I'll try to reproduce that error and report as needed.
